src/matching.rs

```diff
@@ -730,7 +731,9 @@ fn matches_simple_selector<E>(
                          AFFECTED_BY_CHILD_INDEX)
         }
         SimpleSelector::Negation(ref negated) => {
-            negated.iter().any(|s| !matches_simple_selector(s, element, relations))
+            negated.iter().any(|s| {
+                !matches_complex_selector(s, element, parent_bf, relations)
```

Comment:
I think the logic is wrong here. This changed from iterating an "and" chain to an "or" chain (each of item of the latter being an "and")

If I’m getting the logic right, the `!` needs to move to the outside of `any()`. This really needs tests, but without a element tree it’s hard to do in this crate. Please make sure to add tests for this in Servo.